cmd/carbonapi/http/find_handlers.go

```diff
@@ -291,10 +356,11 @@ func findHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if config.Config.ResponseCacheConfig.FindAutoRefreshConfig.Enable {
-		_, ok = queryStore[fmt.Sprint(pv3Request.GetMetrics())]
+		_, ok = queryStore.GetUpdate(fmt.Sprint(pv3Request.GetMetrics()))
 		if !ok {
-			queryAutoRefresh(accessLogger, pv3Request, refreshDelay, lockDelay)
-			queryStore[fmt.Sprint(pv3Request.GetMetrics())] = ""
+			depth := getDepth(pv3Request)
+			go queryAutoRefresh(accessLogger, fmt.Sprint(pv3Request.GetMetrics()), refreshDelay, lockDelay)
```

Comment:
Don't need a go routine
Need to ensure their is a single refresh task for a given key at any time